run-tests.php

```diff
@@ -2027,6 +2028,9 @@ function run_test(string $php, $file, array $env): string
     $extensions = [];
     if ($test->hasSection('EXTENSIONS')) {
         $extensions = preg_split("/[\n\r]+/", trim($test->getSection('EXTENSIONS')));
+        if (in_array("*", $extensions, true)) {
+            $extensions = array_diff($exts_to_test, ["php8apache2_4", "php8phpdbg", "php8ts_debug"]);
```

Comment:
The list of exclusions doesn't belong here; this needs to be handled otherwise. Maybe somebody has an idea (except for using `dl()`).

Reply:
Could probably require the `php_` prefix:

https://github.com/php/php-src/blob/b445641dd71011632fabe2a972d4a592b32d614d/run-tests.php#L869

Or is that an issue on non Windows platforms?

Reply:
Instead of `*` manual list with `?` prefix - loaded only if present?

Reply:
The whole point is to load *all* available extensions (even for external extensions).

Reply:
I mean `*` can be dangerous as it can load unwanted non-php-src extensions.

Reply:
What do you mean by "dangerous"? Security related, or just that even external extensions would be tested as well. If the latter, in my opinion, this is a good thing, since all extensions *should* conform to the contracts established by these tests. If some extensions don't, it will be noticed, and the extensions can be fixed.

Reply:
I mean the latter, if it is wanted to load/run unknown extension at all.